```
Warning: Python 3.10 cannot parse code formatted for Python 3.14. To fix this: run Black with Python 3.14, set --target-version to py310, or use --fast to skip the safety check. Black's safety check verifies equivalence by parsing the AST, which fails when the running Python is older than the target version.
would reformat /home/runner/work/bilingual_book_maker/bilingual_book_maker/tests/test_provider_loader.py

Oh no! 💥 💔 💥
1 file would be reformatted, 35 files would be left unchanged.
Error: Process completed with exit code 1.
```

Seems the ci tool chain failed because of the python version too low for black, the reformatting tool. 

Update: pass now.

--- a/README-CN.md
+++ b/README-CN.md
@@ -165,6 +165,38 @@ bbook --book_name test_books/animal_farm.epub --openai_key ${openai_key} --test
 
 ## 参数说明
 
+- `--model`:
+
+  指定翻译模型。默认：`chatgptapi`。各模型值及其行为：
+
+  | 模型 | Key 来源 | 说明 |
+  |------|---------|------|
+  | `chatgptapi` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-3.5-turbo，自动检测 API 可用模型 |
+  | `gpt4` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-4 系列，自动在可用变体间负载均衡 |
+  | `gpt4omini` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-4o-mini |
+  | `gpt4o` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-4o |
+  | `gpt5mini` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-5-mini |
+  | `o1preview` | `--openai_key` / `BBM_OPENAI_API_KEY` | o1-preview |
+  | `o1` | `--openai_key` / `BBM_OPENAI_API_KEY` | o1 |
+  | `o1mini` | `--openai_key` / `BBM_OPENAI_API_KEY` | o1-mini |
+  | `o3mini` | `--openai_key` / `BBM_OPENAI_API_KEY` | o3-mini |
+  | `openai` | `--openai_key` / `BBM_OPENAI_API_KEY` | **必须配合 `--model_list`**，可使用任意 OpenAI 兼容模型 |
+  | `claude-*` | `--claude_key` / `BBM_CLAUDE_API_KEY` | 前缀匹配，如 `--model claude-sonnet-4-20250514` |
+  | `gemini` | `--gemini_key` / `BBM_GOOGLE_GEMINI_KEY` | Gemini Flash，支持 `--model_list` 自定义 |
+  | `geminipro` | `--gemini_key` / `BBM_GOOGLE_GEMINI_KEY` | Gemini Pro |
+  | `groq` | `--groq_key` / `BBM_GROQ_API_KEY` | **必须配合 `--model_list`** |
+  | `xai` | `--xai_key` / `BBM_XAI_API_KEY` | Grok |
+  | `qwen-mt-turbo` | `--qwen_key` / `BBM_QWEN_API_KEY` | 通义千问快速翻译模型 |
+  | `qwen-mt-plus` | `--qwen_key` / `BBM_QWEN_API_KEY` | 通义千问高质量翻译模型 |
+  | `google` | 无需 key | 免费谷歌翻译 |
+  | `caiyun` | `--caiyun_key` / `BBM_CAIYUN_API_KEY` | 彩云小译 |
+  | `deepl` | `--deepl_key` / `BBM_DEEPL_API_KEY` | DeepL（付费） |
+  | `deeplfree` | 无需 key | DeepL 免费版 |
+  | `tencentransmart` | 无需 key | 腾讯交互翻译，免费 |
+  | `customapi` | `--custom_api` / `BBM_CUSTOM_API` | 自定义翻译 API |
+
+  上表未列出的 OpenAI 兼容 API，请使用 `--provider`（见「自定义 API Provider」章节）。
+
 - `--test`:
 
   如果大家没付费可以加上这个先看看效果（有 limit 稍微有些慢）

--- a/README.md
+++ b/README.md
@@ -179,6 +179,38 @@ bbook --book_name test_books/animal_farm.epub --openai_key ${openai_key} --test
 
 ## Params
 
+- `--model`:
+
+  Select the translation model. Default: `chatgptapi`. Available values and their behavior:
+
+  | Model | Key Source | Notes |
+  |-------|-----------|-------|
+  | `chatgptapi` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-3.5-turbo. Auto-detects available models from API |
+  | `gpt4` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-4 family. Auto-balances across available GPT-4 variants |
+  | `gpt4omini` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-4o-mini |
+  | `gpt4o` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-4o |
+  | `gpt5mini` | `--openai_key` / `BBM_OPENAI_API_KEY` | GPT-5-mini |
+  | `o1preview` | `--openai_key` / `BBM_OPENAI_API_KEY` | o1-preview |
+  | `o1` | `--openai_key` / `BBM_OPENAI_API_KEY` | o1 |
+  | `o1mini` | `--openai_key` / `BBM_OPENAI_API_KEY` | o1-mini |
+  | `o3mini` | `--openai_key` / `BBM_OPENAI_API_KEY` | o3-mini |
+  | `openai` | `--openai_key` / `BBM_OPENAI_API_KEY` | **Requires `--model_list`**. Use any OpenAI-compatible model |
+  | `claude-*` | `--claude_key` / `BBM_CLAUDE_API_KEY` | Prefix match. e.g. `--model claude-sonnet-4-20250514` |
+  | `gemini` | `--gemini_key` / `BBM_GOOGLE_GEMINI_KEY` | Gemini Flash. Supports `--model_list` |
+  | `geminipro` | `--gemini_key` / `BBM_GOOGLE_GEMINI_KEY` | Gemini Pro |
+  | `groq` | `--groq_key` / `BBM_GROQ_API_KEY` | **Requires `--model_list`** |
+  | `xai` | `--xai_key` / `BBM_XAI_API_KEY` | Grok |
+  | `qwen-mt-turbo` | `--qwen_key` / `BBM_QWEN_API_KEY` | Qwen fast translation model |
+  | `qwen-mt-plus` | `--qwen_key` / `BBM_QWEN_API_KEY` | Qwen high-quality translation model |
+  | `google` | N/A | Free. No API key needed |
+  | `caiyun` | `--caiyun_key` / `BBM_CAIYUN_API_KEY` | Caiyun |
+  | `deepl` | `--deepl_key` / `BBM_DEEPL_API_KEY` | DeepL (paid) |
+  | `deeplfree` | N/A | DeepL Free |
+  | `tencentransmart` | N/A | Tencent TranSmart. Free |
+  | `customapi` | `--custom_api` / `BBM_CUSTOM_API` | Custom translation API |
+
+  For any OpenAI-compatible API not listed above, use `--provider` instead (see Custom API Provider section).
+
 - `--test`:
 
   Use `--test` option to preview the result if you haven't paid for the service. Note that there is a limit and it may take some time.

--- a/tests/test_provider_loader.py
+++ b/tests/test_provider_loader.py
@@ -15,7 +15,6 @@ from book_maker.translator.claude_translator import Claude
 from book_maker.translator.gemini_translator import Gemini
 from book_maker.translator.qwen_translator import QwenTranslator
 
-
 VALID_DEEPSEEK = {
     "api_style": "openai",
     "base_url": "https://api.deepseek.com/v1",

